operator/pkg/apis/values_types.proto

```diff
@@ -1387,7 +1392,9 @@ message Values {
   BaseConfig base = 37;
 
   // Configuration for istiod-remote.
-  IstiodRemoteConfig istiodRemote = 38;
+  // DEPRECATED - istiod-remote chart is removed and replaced with
+  // `istio-discovery --set values.istiodRemote.enabled=true`
+  IstiodRemoteConfig istiodRemote = 38 [deprecated = true];
```

Comment:
I'm not as familiar with how deprecation works with helm charts - should we mark it as deprecated in one release (1.24) but actually remove the chart in a subsequent release (1.25). That way in case folks were leveraging the istiod-remote helm charts they would have time to migrate to the istiod remote profile prior to the chart being removed?

@howardjohn

Reply:
As per WG, since we have never publicly documented the Helm chart `istiod-remote`, we are OK to remove it for 1.24 and document the correct way: https://docs.google.com/document/d/1wsa06GGiq1LEGwhkiPP0FKIZJqdAiue-VeBonWAzAyk/edit

an upgradeNote in the release notes for this PR should be enough.